thorlcr/activities/diskread/thdiskreadslave.cpp

```diff
@@ -250,46 +256,95 @@ void CDiskRecordPartHandler::open()
         activity.totalProgress += partStream->queryProgress();
     partStream.clear();
 
-    unsigned rwFlags = DEFAULT_RWFLAGS;
+    unsigned rwFlags = 0;
     if (checkFileCrc) // NB: if compressed, this will be turned off by base class
         rwFlags |= rw_crc;
     if (activity.grouped)
         rwFlags |= rw_grouped;
 
+    IOutputMetaData *projectedFormat = activity.helper->queryProjectedDiskRecordSize();
+    IOutputMetaData *expectedFormat = activity.helper->queryDiskRecordSize();
     Owned<ITranslator> translator = activity.getTranslators(*partDesc);
-    if (compressed)
+    IOutputMetaData *actualFormat = translator ? &translator->queryActualFormat() : expectedFormat;
+    bool canSerializeTypeInfo = actualFormat->queryTypeInfo()->canSerialize() && projectedFormat->queryTypeInfo()->canSerialize();
+    if (canSerializeTypeInfo)
     {
-        rwFlags |= rw_compress;
-        partStream.setown(createRowStream(iFile, activity.queryProjectedDiskRowInterfaces(), rwFlags, activity.eexp, translator, this));
-        if (!partStream.get())
+        for (unsigned copy=0; copy<partDesc->numCopies(); copy++)
         {
-            if (!blockCompressed)
-                throw MakeStringException(-1,"Unsupported compressed file format: %s", filename.get());
-            else
-                throw MakeActivityException(&activity, 0, "Failed to open block compressed file '%s'", filename.get());
+            RemoteFilename rfn;
+            partDesc->getFilename(copy, rfn);
+
+            StringBuffer path;
+            if (!rfn.isLocal() || testForceRemote(rfn.getLocalPath(path)))
+            {
+                // Open a stream from remote file, having passed actual, expected, projected, and filters to it
+                SocketEndpoint ep(rfn.queryEndpoint());
+                setDafsEndpointPort(ep);
+
+                RowFilter actualFilter;
+                Owned<const IKeyTranslator> keyedTranslator = createKeyTranslator(actualFormat->queryRecordAccessor(true), expectedFormat->queryRecordAccessor(true));
```

Comment:
could avoid creating the translator if activity.fieldFilters is empty.

Reply:
( also if actual = projected, but pointers are unlikely to be equal at the moment...)

Reply:
ok, have pushed a commit a avoid if not necessary.